source/MRViewer/MRModalDialog.cpp

```diff
@@ -0,0 +1,115 @@
+#include "MRModalDialog.h"
+
+#include "ImGuiHelpers.h"
+#include "MRRibbonConstants.h"
+#include "MRRibbonFontManager.h"
+#include "MRUIStyle.h"
+
+namespace MR
+{
+
+ModalDialog::ModalDialog( std::string label, ModalDialogSettings settings )
+    : label_( std::move( label ) )
+    , settings_( std::move( settings ) )
+{
+}
+
+bool ModalDialog::beginPopup( float menuScaling )
+{
+    const auto windowWidth = settings_.windowWidth > 0.f ? settings_.windowWidth : cModalWindowWidth * menuScaling;
+    const ImVec2 windowSize { windowWidth, -1 };
+    ImGui::SetNextWindowSize( windowSize, ImGuiCond_Always );
+
+    const ImVec2 windowPadding { cModalWindowPaddingX * menuScaling, cModalWindowPaddingY * menuScaling };
+    const ImVec2 itemSpacing { 2.0f * cDefaultItemSpacing * menuScaling, 3.0f * cDefaultItemSpacing * menuScaling };
+    const ImVec2 itemInnerSpacing { 2.0f * cDefaultInnerSpacing * menuScaling, cDefaultInnerSpacing * menuScaling };
+    ImGui::PushStyleVar( ImGuiStyleVar_WindowPadding, windowPadding );
+    ImGui::PushStyleVar( ImGuiStyleVar_ItemSpacing, itemSpacing );
+    ImGui::PushStyleVar( ImGuiStyleVar_ItemInnerSpacing, itemInnerSpacing );
+
+    const ImGuiWindowFlags flags = ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove | ImGuiWindowFlags_AlwaysAutoResize | ImGuiWindowFlags_NoTitleBar;
+    if ( !ImGui::BeginModalNoAnimation( label_.c_str(), nullptr, flags ) )
+    {
+        ImGui::PopStyleVar(); // ImGuiStyleVar_ItemSpacing
```

Comment:
why not `PopStyleVar(2)`

Reply:
actually 3 needed in for `ImGuiStyleVar_ItemInnerSpacing`

Reply:
Fixed.